src/graph/optimizer/OptGroup.cpp

```diff
@@ -41,9 +41,81 @@ OptGroup::OptGroup(OptContext *ctx) noexcept : ctx_(ctx) {
   DCHECK(ctx != nullptr);
 }
 
+Status OptGroup::validateSubPlan(const OptGroupNode *gn,
+                                 const OptRule *rule,
+                                 const std::vector<OptGroup *> &patternLeaves) const {
+  auto &deps = DCHECK_NOTNULL(gn)->dependencies();
+
+  auto checkDepGroup = [this, gn, rule, &patternLeaves](const OptGroup *depGroup) -> Status {
+    auto iter = std::find(patternLeaves.begin(), patternLeaves.end(), depGroup);
+    if (iter != patternLeaves.end()) {
+      return Status::OK();
+    }
+    if (!depGroup) {
+      return Status::Error("Could not find the dependent group in pattern leaves");
+    }
+    if (depGroup->groupNodes_.size() != 1U || depGroup->groupNodesReferenced_.size() != 1U) {
+      return Status::Error(
+          "Invalid sub-plan generated when applying the rule: %s, "
+          "planNode: %s, numGroupNodes: %lu, numGroupNodesRef: %lu",
+          rule->toString().c_str(),
+          PlanNode::toString(gn->node()->kind()),
+          depGroup->groupNodes_.size(),
+          depGroup->groupNodesReferenced_.size());
+    }
```

Comment:
It is better not to pass optimizer errors to the user.
It might change this way: 
```
DCHECK_NOTNULL(depGroup) << "Could not find the dependent group in pattern leaves";
```
or 
```
// The `depGroup` should not be null bcz ...
DCHECK_NOTNULL(depGroup);
```

Reply:
I have converted the error message in QueryInstance. this return only for not crash in release version.

Reply:
Ok.